src/main/webapp/app/entities/theory/theory.component.html

```diff
@@ -1,67 +1,47 @@
-                        </button>
-                        <button type="submit"
-                                [routerLink]="['/theory', { outlets: { popup: theory.id + '/delete'} }]"
-                                replaceUrl="true"
-                                queryParamsHandling="merge"
-                                class="btn btn-danger btn-sm">
-                            <fa-icon [icon]="'times'"></fa-icon>
-                            <span class="d-none d-md-inline" jhiTranslate="entity.action.delete">Delete</span>
-                        </button>
+
+    <div class = "container col-offset-1">
+        <div class="row">
+            <div class="input-group mb-3 col">
+                <input [(ngModel)]="searchValue" (ngModelChange)="onSearchChange()" class="form-control" placeholder="{{'entity.action.liveSearch'| translate}}" type="text"> </div>
+            <div class="input-group mb-3 col-3">
+                <select [(ngModel)]="categoryId" (ngModelChange)="onSearchChange()" style = "width: 100%;" class="custom-select" id="inputGroupSelect01">
+                    <option selected value="" jhiTranslate="learnportalApp.category.choose"></option>
+                    <option *ngFor="let category of categories ;trackBy: trackId" value="{{category.id}}">{{category.shortName}}</option>
+                </select>
+            </div>
+            <div *jhiHasAnyAuthority="'ROLE_ADMIN'" class=" col-2">
+                <button [routerLink]="['/theory/new']" style = "width: 100%;" class="btn btn-primary" type="button"><fa-icon [icon]="'plus'"></fa-icon> Добавить</button>
+            </div>
+        </div>
+
+        <div class="alert alert-warning" *ngIf="theories?.length === 0">
+            <span jhiTranslate="learnportalApp.theory.home.notFound"></span>
+        </div>
+
+        <div infinite-scroll (scrolled)="loadPage(page + 1)" [infiniteScrollDisabled]="page >= links['last']" [infiniteScrollDistance]="0">
+            <div style = "margin-top: 30px;" class="card" *ngFor="let theory of theories ;trackBy: trackId">
+                <h5 style="background-color: #708090;" class="card-header theory-link">
+                    <a class="theory-link" [routerLink]="['/theory', theory.id, 'view' ]">
+                        <div class="row">
+                            <div title="{{theory.title}}" style="color: white;" class="text-reduction col-6">{{theory.title}}</div>
+                            <div title="{{theory.category?.shortName}}" style="color: #CDCED0;" class="text-reduction col-6 text-right">{{theory.category?.shortName}}</div>
+                        </div>
+                    </a>
+                </h5>
+                <div class="card-body">
+                    <p title="{{theory.description}}" class="text-reduction card-text">{{theory.description}}</p>
+                    <div *jhiHasAnyAuthority="'ROLE_ADMIN'" class="text-right">
```

Comment:
Do you really need double quotes and then single quotes?

Reply:
yes

Reply:
Why?

Reply:
double quotes are used for escape single quotes, I tested we need both

Reply:
Ok, discussed